docs/catalog/cloud/use-compiled-languages.md

```diff
@@ -0,0 +1,44 @@
+---
+version: 1.0
+submitted_by: markus-ntt-seidl
+published_date: TBD
+category: cloud
+tags: 
+ - cloud
+ - role:cloud-engineer
+ - size:small
+---
+
+# Use compiled languages
+
+## Description
+
+Interpreted languages need to be parsed, compiled and executed when the application starts or a workload arrives. This tends to be more energy heavy then when a compiled language is used. The compilation is then only done once, saving on resources.
+
+## Solution
+
+Use compiled languages whenever possible or compile interpreted languages.
+
+## SCI Impact
+
+`SCI = (E * I) + M per R`  
+[Software Carbon Intensity Spec](https://grnsft.org/sci)
+
+Concerning the SCI equation, use compiled languages will impact as follows:
+
+- `E`: The embodied carbon emissions will be reduced, as compiled binaries tend to be smaller then the sources they are compiled from
+- `I`: Running compiled binaries is more energy efficient and uses less energy which outweighs the energy consumed in compiling it to binary upfront.
+
+## Assumptions
+
+- There is the assumption that the execution environment allows usage of compiled languages, this is not always the case (for example web browsers)
+- Use benchmarks to determine if the application use case benefits from using a compiled language
+
+## Considerations
+
+- Some interpreted languages provide ways to compile these into binary (for example GraalVM for Java) 
```

Comment:
Maybe this is covered in the benchmarks item above, but there is a tradeoff based on how frequently the code needs to be complied. For example, codebases that are rarely changed and heavily used would have a much larger impact from being a compiled language versus ones that are frequently changed and have low usage. The interpreted language can be a benefit if the compilation impact is large.

Reply:
I feel that this whole pattern is a rabbit hole where you can go down on every point.

Compile Time (to binary) vs. Execution Time per Request vs. Compile Time (when dynamically compiled upon a request, as languages do cache) vs. Multi-Stage Compilers (compile fast first, optimize hot paths like JavaScript/NodeJS does) vs. Serverless (which might be more compiled on request as they are shut down in between) vs ...